src/test/groovy/org/jenkinsci/gradle/plugins/jpi2/V2IntegrationTest.java

```diff
@@ -948,6 +948,68 @@ public PluginAction(String name) {
                 .containsExactlyInAnyOrder("test-plugin-1.0.0.jar");
     }
 
+    @Test
+    void getsCorrectGuiceVersion() throws IOException {
+        // given
+        var ith = new IntegrationTestHelper(tempDir);
+        initBuild(ith);
+        Files.write((getBasePluginConfig() +/* language=kotlin */ """
+                dependencies {
+                    annotationProcessor("org.projectlombok:lombok:1.18.38")
+                    compileOnly("org.projectlombok:lombok:1.18.38")
+                    runtimeOnly("com.google.inject:guice:5.1.0") // This is an older version of Guice that should get upgraded
+                }
+                configurations.getByName("compileClasspath").shouldResolveConsistentlyWith(configurations.getByName("runtimeClasspath"))
+                """).getBytes(StandardCharsets.UTF_8), ith.inProjectDir("build.gradle.kts"));
+        ith.mkDirInProjectDir("src/main/java/com/example/plugin");
+        Files.write(/* language=java */ """
+                package com.example.plugin;
+                import lombok.*;
+                import hudson.Extension;
+                import jakarta.inject.Inject;
+                import hudson.model.RootAction;
+                @Extension
+                @NoArgsConstructor
+                public class PluginAction implements RootAction {
+                    private String name;
+                    @Inject
+                    public PluginAction(String name) {
+                        this.name = name;
+                    }
+                    public String getUrlName() { return "example"; }
+                    public String getDisplayName() { return "Example plugin"; }
+                    public String getIconFileName() { return null; }
+                }
+                """.getBytes(StandardCharsets.UTF_8), ith.inProjectDir("src/main/java/com/example/plugin/PluginAction.java"));
+
+        // when
+        var gradleRunner = ith.gradleRunner();
+        var result = gradleRunner.withArguments("build").build();
+
+        // then
+        assertThat(result.getOutput()).contains("BUILD SUCCESSFUL");
+
+        var explodedWar = ith.inProjectDir("build/jpi");
+
+        var jpiLibsDir = new File(explodedWar, "WEB-INF/lib");
+        assertThat(jpiLibsDir).exists();
+
+        var jpiLibs = jpiLibsDir.list();
+        assertThat(jpiLibs).isNotNull()
+                .containsExactlyInAnyOrder("aopalliance-1.0.jar",
+                        "jakarta.inject-api-2.0.1.jar",
+                        "guice-6.0.0.jar",
```

Comment:
We probably need another change to drop dependencies that are already part of jenkins-core's tree.

Reply:
We have a task that excludes anything present in core or another plugin that could be used as reference